gen/build_deploy/bash.py

```diff
@@ -701,24 +707,26 @@ def do_create(tag, build_name, reproducible_artifact_path, commit, variant_argum
     """
     # TODO(cmaloney): Build installers in parallel.
     # Variants are sorted for stable ordering.
-    for variant, bootstrap_info in sorted(variant_arguments.items(),
-                                          key=lambda kv: pkgpanda.util.variant_str(kv[0])):
-        with logger.scope("Building installer for variant: ".format(pkgpanda.util.variant_name(variant))):
-            bootstrap_installer_name = '{}installer'.format(pkgpanda.util.variant_prefix(variant))
-            installer_filename = make_installer_docker(variant, all_completes[variant],
-                                                       all_completes[bootstrap_installer_name])
-
-            yield {
-                'channel_path': 'dcos_generate_config.{}sh'.format(pkgpanda.util.variant_prefix(variant)),
-                'local_path': installer_filename
-            }
+    for variant in sorted(variant_arguments.keys(), key=lambda k: pkgpanda.util.variant_str(k)):
+        variant_name = pkgpanda.util.variant_name(variant)
+        bootstrap_installer_name = '{}installer'.format(pkgpanda.util.variant_prefix(variant))
```

Comment:
That doesn't convince me of anything. That is a good time to use a template because it is a template. I.E. a data structure parameterized for a bunch of variables. Your source is separate from your structure; you need to be explicit with your templating. You also have more than one varaible

Now compare that to here. I am constructing a string with ONE parameter exactly ONCE to use it ONCE. As I see it, you are just asking me to take one line and split it into 2 so that I am defining what my string *could* look like on one line and then how it actually looks on another. That doesnt seem much more readable to me.

In any case, as I already stated, that is not in this PR and if you want to see that changed, take this over to https://github.com/dcos/dcos/pull/1353. This is a PR for the Azure support in dcos-launch